src/class-rest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * REST API class
+ *
+ * @package Parsely
+ * @since 3.1.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+use WP_Post;
+
+/**
+ * Injects Parse.ly Metadata to WordPress REST API
+ *
+ * @since 3.1.0
+ */
+class Rest {
+	private const PARSELY_REST_VERSION    = '1.0.0';
+	private const PARSELY_REST_FIELD_NAME = 'parsely';
+
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Register fields in WordPress REST API
+	 *
+	 * @since 3.1.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		/**
+		 * Filter whether REST API support is enabled or not.
+		 *
+		 * @since 3.1.0
+		 *
+		 * @param bool $enabled True if enabled, false if not.
+		 */
+		if ( apply_filters( 'wp_parsely_enable_rest_api_support', true ) ) {
+			add_action( 'rest_api_init', array( $this, 'register_parsely_meta' ) );
+		}
+	}
+
+	/**
+	 * Registers the `parsely-meta` field in the REST API.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @return void
+	 */
+	public function register_parsely_meta(): void {
+		$callback = function( array $object ): array {
```

Comment:
Is there a reason this shouldn't be a class member function instead of being declared inside this one?